Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/FRFS/OnCancel.hs

```diff
@@ -59,19 +59,18 @@ onCancel _ booking' dOnCancel = do
     Spec.ON_CANCEL_SOFT_CANCEL -> do
       void $ QTBooking.updateRefundCancellationChargesAndIsCancellableByBookingId (Just dOnCancel.refundAmount) (Just dOnCancel.cancellationCharges) (Just True) booking.id
     Spec.ON_CANCEL_CANCELLED -> do
-      void $ checkRefundAndCancellationCharges booking.id dOnCancel
-      void $ QTBooking.updateStatusById FTBooking.CANCELLED booking.id
-      void $ QTicket.updateAllStausByBookingId DFRFSTicket.CANCELLED booking.id
-      void $ QTBP.updateStatusByTicketBookingId DTBP.REFUND_PENDING booking.id
-      void $ QTBooking.updateIsBookingCancellableByBookingId (Just True) booking.id
-  return ()
-  where
-    checkRefundAndCancellationCharges bookingId onCancelReq = do
-      booking <- runInReplica $ QTBooking.findById bookingId >>= fromMaybeM (BookingDoesNotExist bookingId.getId)
+      -- booking <- runInReplica $ QTBooking.findById booking.id >>= fromMaybeM (BookingDoesNotExist booking.id.getId)
       case booking of
         Booking.FRFSTicketBooking {refundAmount = Just rfAmount, cancellationCharges = Just charges} -> do
-          when (rfAmount /= onCancelReq.refundAmount) $
+          when (rfAmount /= dOnCancel.refundAmount) $
             throwError $ InternalError "Refund Amount mismatch in onCancel Req"
-          when (charges /= onCancelReq.cancellationCharges) $
+          when (charges /= dOnCancel.cancellationCharges) $
             throwError $ InternalError "Cancellation Charges mismatch in onCancel Req"
-        _ -> throwError $ InternalError "Refund Amount or Cancellation Charges not found in booking"
+          void $ QTBooking.updateStatusById FTBooking.CANCELLED booking.id
+          void $ QTicket.updateAllStausByBookingId DFRFSTicket.CANCELLED booking.id
+          void $ QTBP.updateStatusByTicketBookingId DTBP.REFUND_PENDING booking.id
+          void $ QTBooking.updateIsBookingCancellableByBookingId (Just True) booking.id
+        _ -> do
+          void $ QTBooking.updateStatusById FTBooking.COUNTER_CANCELLED booking.id
```

Comment:
Do we get counter cancelled case in on_cancel or in on_status?

Reply:
In case of cancellation, have you verified we don't get refund amount and cancellation charges?